vpr/src/route/rr_graph.cpp

```diff
@@ -1241,6 +1261,20 @@ static void build_rr_graph(const t_graph_type graph_type,
     }
     /* END SB LOOKUP */
 
+    /* Add extra nodes to switch blocks for multi-layer FPGAs with custom switch blocks that define track-to_track connections */
+    vtr::NdMatrix<t_inter_die_switchblock_edge, 5> multi_layer_track_conn; //keeps the extra nodes offset and drivers for each destination track
```

Comment:
Needs a big comment defining dimensions so we can see how to access. E.g. [0..grid.layers-1][0..grid.width-1][grid.height-1][0..max_chan_width-1][CHANX or CHANY].